activestorage/lib/active_storage/service/s3_service.rb

```diff
@@ -16,6 +16,8 @@ class Service::S3Service < Service
 
     def initialize(bucket:, upload: {}, public: false, **options)
       @client = Aws::S3::Resource.new(**options)
+      @s3_client = @client.client
+      @transfer_manager = Aws::S3::TransferManager.new(client: @s3_client) if defined?(Aws::S3::TransferManager)
```

Comment:
Are we sure that transfert manager can be shared? Doesn't it have some internal state or something that may not be thread safe?

Reply:
Answering my own question. I went into the gem. The client have a pool of connection, so seems thread safe, the manager has basically no state. So yes, we can share all that.

Reply:
Yes you beat me to it. It essentially just wraps an `Aws::S3::Client` which is thread safe (and already used here by the `Aws::S3::Resource`.